From c4cc546c778827f61b8c88282cb7582a5724f833 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Fri, 13 Nov 2020 05:56:46 -0500
Subject: [PATCH] move from product

Change-Id: If59377d4c08405e3981c12cfc1a4d4d854149567
---
 Android.bp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Android.bp b/Android.bp
index a19761b..5f10da0 100644
--- a/Android.bp
+++ b/Android.bp
@@ -15,5 +15,4 @@
 prebuilt_font {
     name: "ArbutusSlab-Regular.ttf",
     src: "ArbutusSlab-Regular.ttf",
-    product_specific: true,
 }
-- 
2.17.1

