From ba2005be0685ee7a3c0345172cdd9f7a4f2dbe3a Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 24 Mar 2021 08:15:18 -0400
Subject: [PATCH 147/230] libs: hwui: switch to -O3

---
 libs/hwui/Android.bp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libs/hwui/Android.bp b/libs/hwui/Android.bp
index 747ce55f498..f15e3830719 100644
--- a/libs/hwui/Android.bp
+++ b/libs/hwui/Android.bp
@@ -168,6 +168,7 @@ cc_library_headers {
 cc_defaults {
     name: "android_graphics_apex",
     cflags: [
+        "-O3",
         "-Wno-unused-parameter",
         "-Wno-non-virtual-dtor",
         "-Wno-maybe-uninitialized",
@@ -179,7 +180,7 @@ cc_defaults {
         "-Wunreachable-code",
     ],
 
-    cppflags: ["-Wno-conversion-null"],
+    cppflags: ["-Wno-conversion-null", "-O3"],
 
     srcs: [
         "apex/android_matrix.cpp",
-- 
2.25.1

