From cd208df019a91b41879d38368d8fddc2e034b335 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sat, 27 Dec 2025 10:24:27 +0100
Subject: [PATCH 13/22] huawei: add additional gralloc 10 usage bits need by L1
 DRM Widevine emui12

---
 board-base.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/board-base.mk b/board-base.mk
index 8cde64f..f409c8a 100644
--- a/board-base.mk
+++ b/board-base.mk
@@ -12,3 +12,6 @@ BUILD_BROKEN_ELF_PREBUILT_PRODUCT_COPY_FILES := true
 BOARD_ROOT_EXTRA_SYMLINKS := $(filter-out $(BOARD_ROOT_EXTRA_SYMLINKS),/mnt/vendor/persist:/persist)
 
 TARGET_NO_KERNEL_OVERRIDE := true
+
+# Graphics
+TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS := 0x4000000
-- 
2.25.1

