From e7857e62f40da706d1a702421f3622399ccd780b Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Fri, 6 Dec 2024 08:17:36 +0100
Subject: [PATCH 16/21] ims: add foss phh ims

---
 base.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/base.mk b/base.mk
index ff6a7cf..b61e32f 100644
--- a/base.mk
+++ b/base.mk
@@ -259,6 +259,10 @@ PRODUCT_PACKAGES += \
 	Iwlan \
 	QualifiedNetworksService
 
+# IMS
+PRODUCT_PACKAGES += \
+    PhhIMS
+
 # Two-pane layout in Settings
 PRODUCT_PACKAGES += \
     androidx.window.extensions
-- 
2.25.1

