From 8793695f6312ee093435422a933b13994452462d Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sat, 27 Dec 2025 09:57:38 +0100
Subject: [PATCH 17/18] huawei: add nfc and nfc.hce stuff

---
 base.mk      | 18 ++++++++++--------
 rw-system.sh | 22 ++++------------------
 2 files changed, 14 insertions(+), 26 deletions(-)

diff --git a/base.mk b/base.mk
index bafaf68..e07ca52 100644
--- a/base.mk
+++ b/base.mk
@@ -6,7 +6,10 @@ PRODUCT_COPY_FILES := \
 	frameworks/native/data/etc/android.hardware.telephony.ims.xml:system/etc/permissions/android.hardware.telephony.ims.xml \
 	frameworks/native/data/etc/android.hardware.bluetooth.xml:system/etc/permissions/android.hardware.bluetooth.xml \
 	frameworks/native/data/etc/android.hardware.bluetooth_le.xml:system/etc/permissions/android.hardware.bluetooth_le.xml \
-	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
+	frameworks/native/data/etc/android.hardware.nfc.hce.xml:system/etc/permissions/android.hardware.nfc.hce.xml \
+	frameworks/native/data/etc/android.hardware.nfc.hcef.xml:system/etc/permissions/android.hardware.nfc.hcef.xml \
+	frameworks/native/data/etc/android.hardware.nfc.xml:system/etc/permissions/android.hardware.nfc.xml \
+	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml
 
 # Bluetooth Audio (System-side HAL, sysbta)
 PRODUCT_PACKAGES += \
@@ -62,16 +65,15 @@ PRODUCT_COPY_FILES += \
 	frameworks/av/services/audiopolicy/config/usb_audio_policy_configuration.xml:system/etc/usb_audio_policy_configuration.xml \
 	device/phh/treble/files/fake_audio_policy_volume.xml:system/etc/fake_audio_policy_volume.xml \
 
-# NFC:
-#   Provide default libnfc-nci.conf file for devices that does not have one in
-#   vendor/etc
-PRODUCT_COPY_FILES += \
-	device/phh/treble/nfc/libnfc-nci.conf:system/phh/libnfc-nci-oreo.conf \
-	device/phh/treble/nfc/libnfc-nci-huawei.conf:system/phh/libnfc-nci-huawei.conf
+# NFC (Huawei):
+#   Default libnfc-nci.conf file for devices that does not have one
+#   in vendor/etc or in /odm/etc/ need copy it in post install
 
 # LineageOS build may need this to make NFC work
 PRODUCT_PACKAGES += \
-        NfcNci \
+	NfcNci \
+	Tag \
+	com.android.nfc_extras
 
 PRODUCT_COPY_FILES += \
 	device/phh/treble/rw-system.sh:system/bin/rw-system.sh \
diff --git a/rw-system.sh b/rw-system.sh
index 107c654..1e133c2 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -584,10 +584,6 @@ if [ -f /vendor/bin/hw/vendor.samsung.hardware.miscpower@1.0-service ] && [ "$vn
     mount -o bind /system/phh/empty /vendor/bin/hw/android.hardware.power@1.0-service
 fi
 
-if [ "$vndk" = 27 ] || [ "$vndk" = 26 ]; then
-    mount -o bind /system/phh/libnfc-nci-oreo.conf /system/etc/libnfc-nci.conf
-fi
-
 if busybox_phh unzip -p /vendor/app/ims/ims.apk classes.dex | grep -qF -e Landroid/telephony/ims/feature/MmTelFeature -e Landroid/telephony/ims/feature/MMTelFeature; then
     mount -o bind /system/phh/empty /vendor/app/ims/ims.apk
 fi
@@ -596,10 +592,6 @@ if getprop ro.hardware | grep -qF exynos; then
     setprop debug.sf.latch_unsignaled 1
 fi
 
-if getprop ro.product.model | grep -qF ANE; then
-    setprop debug.sf.latch_unsignaled 1
-fi
-
 if getprop ro.vendor.product.device | grep -q -e nora -e rhannah; then
     setprop debug.sf.latch_unsignaled 1
 fi
@@ -640,23 +632,17 @@ if getprop ro.vendor.build.fingerprint | grep -iq -e Redmi/rosemary \
     mount -o bind /system/phh/rosemary-excluded-input-devices.xml /system/etc/excluded-input-devices.xml
 fi
 
+# Iceows patch - NFC NXP conf remove, you must set by specify model patch
+# remove gnss_watchlssd_thirdparty to avoid log spam
 if getprop ro.vendor.build.fingerprint | grep -iq -E -e 'huawei|honor' || getprop persist.sys.overlay.huawei | grep -iq -E -e 'true'; then
-    p=/product/etc/nfc/libnfc_nxp_*_*.conf
-    mount -o bind "$p" /system/etc/libnfc-nxp.conf ||
-        mount -o bind /product/etc/libnfc-nxp.conf /system/etc/libnfc-nxp.conf || true
-
-    p=/product/etc/nfc/libnfc_brcm_*_*.conf
-    mount -o bind "$p" /system/etc/libnfc-brcm.conf ||
-        mount -o bind /product/etc/libnfc-nxp.conf /system/etc/libnfc-nxp.conf || true
-
-    mount -o bind /system/phh/libnfc-nci-huawei.conf /system/etc/libnfc-nci.conf
+    setprop debug.sf.latch_unsignaled 1
 
     # Set proper /sdcard permissions to avoid dead storage
     chown media_rw:media_rw /data/media/0
     chmod 0770 /data/media/0
 
     # Disable this watchlss extension to fix logspams and dead gps
-    mount -o bind /mnt/phh/empty /vendor/bin/gnss_watchlssd_thirdparty
+    mount -o bind /system/phh/empty /vendor/bin/gnss_watchlssd_thirdparty
 
     # For audio_custom
     chown system:system /sys/class/sensors/rpc_sensor/rpc_motion_req
-- 
2.25.1

