From 3c1839c49a8385c101d5b60c13e936dde02092ee Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 19 Dec 2020 05:34:34 -0500
Subject: [PATCH 10/80] rework securize tweak

---
 rw-system.sh | 68 +++++++++++++++++++++++++++-------------------------
 1 file changed, 35 insertions(+), 33 deletions(-)

diff --git a/rw-system.sh b/rw-system.sh
index 3f6efdd..615a371 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -676,40 +676,42 @@ fi
 
 setprop ctl.stop console
 dmesg -n 1
+
+copyprop() {
+    p="$(getprop "$2")"
+    if [ "$p" ]; then
+        resetprop_phh "$1" "$(getprop "$2")"
+    fi
+}
+
+copyprop ro.build.device ro.vendor.build.device
+copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
+copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
+copyprop ro.build.fingerprint ro.vendor.build.fingerprint
+copyprop ro.build.device ro.vendor.product.device
+copyprop ro.product.system.device ro.vendor.product.device
+copyprop ro.product.device ro.vendor.product.device
+copyprop ro.product.system.device ro.product.vendor.device
+copyprop ro.product.device ro.product.vendor.device
+copyprop ro.product.system.name ro.vendor.product.name
+copyprop ro.product.name ro.vendor.product.name
+copyprop ro.product.system.name ro.product.vendor.name
+copyprop ro.product.name ro.product.vendor.name
+copyprop ro.system.product.brand ro.vendor.product.brand
+copyprop ro.product.brand ro.vendor.product.brand
+copyprop ro.product.system.model ro.vendor.product.model
+copyprop ro.product.model ro.vendor.product.model
+copyprop ro.product.system.model ro.product.vendor.model
+copyprop ro.product.model ro.product.vendor.model
+copyprop ro.build.product ro.vendor.product.model
+copyprop ro.build.product ro.product.vendor.model
+copyprop ro.system.product.manufacturer ro.vendor.product.manufacturer
+copyprop ro.product.manufacturer ro.vendor.product.manufacturer
+copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
+copyprop ro.product.manufacturer ro.product.vendor.manufacturer
+copyprop build.version.security_patch ro.build.version.security_patch
+
 if [ -f /system/phh/secure ];then
-    copyprop() {
-        p="$(getprop "$2")"
-        if [ "$p" ]; then
-            resetprop_phh "$1" "$(getprop "$2")"
-        fi
-    }
-
-    copyprop ro.build.device ro.vendor.build.device
-    copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
-    copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
-    copyprop ro.build.fingerprint ro.vendor.build.fingerprint
-    copyprop ro.build.device ro.vendor.product.device
-    copyprop ro.product.system.device ro.vendor.product.device
-    copyprop ro.product.device ro.vendor.product.device
-    copyprop ro.product.system.device ro.product.vendor.device
-    copyprop ro.product.device ro.product.vendor.device
-    copyprop ro.product.system.name ro.vendor.product.name
-    copyprop ro.product.name ro.vendor.product.name
-    copyprop ro.product.system.name ro.product.vendor.name
-    copyprop ro.product.name ro.product.vendor.name
-    copyprop ro.system.product.brand ro.vendor.product.brand
-    copyprop ro.product.brand ro.vendor.product.brand
-    copyprop ro.product.system.model ro.vendor.product.model
-    copyprop ro.product.model ro.vendor.product.model
-    copyprop ro.product.system.model ro.product.vendor.model
-    copyprop ro.product.model ro.product.vendor.model
-    copyprop ro.build.product ro.vendor.product.model
-    copyprop ro.build.product ro.product.vendor.model
-    copyprop ro.system.product.manufacturer ro.vendor.product.manufacturer
-    copyprop ro.product.manufacturer ro.vendor.product.manufacturer
-    copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
-    copyprop ro.product.manufacturer ro.product.vendor.manufacturer
-    copyprop build.version.security_patch ro.build.version.security_patch
     (getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
         [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
     done
-- 
2.25.1

