From b0822ab64a42d1da26a86ed3377968edc9b0836d Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Mon, 8 Mar 2021 04:46:57 -0500
Subject: [PATCH 19/80] tweaks for UMIDIGI A3X & A9 (overlay.devinputjack)

---
 rw-system.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/rw-system.sh b/rw-system.sh
index 615a371..0230045 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -389,6 +389,11 @@ if getprop ro.vendor.build.fingerprint |grep -qi -e Redmi/merlin -e Redmi/lancel
     setprop persist.sys.overlay.devinputjack true
 fi
 
+# UMIDIGI/A3X & UMIDIGI/A9
+if getprop ro.vendor.build.fingerprint |grep -qi -e UMIDIGI/A3X -e UMIDIGI/A9; then
+    setprop persist.sys.overlay.devinputjack true
+fi
+
 #Realme 6
 if getprop ro.vendor.product.device |grep -iq -e RMX2001 -e RMX2151 -e RMX2111 -e RMX2111L1;then
     setprop persist.sys.phh.fingerprint.nocleanup true
-- 
2.25.1

