From 41abc065bbd8116d595d134ed967086f7c72c6d9 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Mon, 8 Mar 2021 04:50:14 -0500
Subject: [PATCH 20/80] tweak for CUBOT KINGKONG MINI2 (overlay.devinputjack)

---
 rw-system.sh | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index 0230045..0b22f42 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -389,11 +389,16 @@ if getprop ro.vendor.build.fingerprint |grep -qi -e Redmi/merlin -e Redmi/lancel
     setprop persist.sys.overlay.devinputjack true
 fi
 
-# UMIDIGI/A3X & UMIDIGI/A9
+# UMIDIGI A3X & A9
 if getprop ro.vendor.build.fingerprint |grep -qi -e UMIDIGI/A3X -e UMIDIGI/A9; then
     setprop persist.sys.overlay.devinputjack true
 fi
 
+# CUBOT KINGKONG MINI2
+if getprop ro.vendor.build.fingerprint |grep -qi -e CUBOT/KINGKONG_MINI2; then
+    setprop persist.sys.overlay.devinputjack true
+fi
+
 #Realme 6
 if getprop ro.vendor.product.device |grep -iq -e RMX2001 -e RMX2151 -e RMX2111 -e RMX2111L1;then
     setprop persist.sys.phh.fingerprint.nocleanup true
-- 
2.25.1

