From 06e413d908da3c3abc5644c6ea238eaf714690ee Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 1 May 2021 10:48:10 -0400
Subject: [PATCH 29/80] allow Wellbeing for gapps

---
 overrides/Android.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/overrides/Android.mk b/overrides/Android.mk
index d50ee9d..9af38cf 100644
--- a/overrides/Android.mk
+++ b/overrides/Android.mk
@@ -3,7 +3,6 @@ LOCAL_PATH := $(call my-dir)
 include $(CLEAR_VARS)
 LOCAL_MODULE := phh-gapps-go-overrides
 PACKAGES.$(LOCAL_MODULE).OVERRIDES := \
-	Wellbeing \
 	GMailGo \
 	YouTubeGo \
 	DuoGo \
@@ -53,7 +52,6 @@ PACKAGES.$(LOCAL_MODULE).OVERRIDES := \
 	Videos \
 	Duo \
 	Traceur \
-	Wellbeing \
 	YouTube \
 	Photos \
 	Maps \
-- 
2.25.1

