From 90a66b3c5bfdb22f1a3bad45291568ac82163114 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Thu, 13 May 2021 13:29:00 -0400
Subject: [PATCH 31/80] try to set building non-sparse image

---
 board-base.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board-base.mk b/board-base.mk
index 4739954..2d039ee 100644
--- a/board-base.mk
+++ b/board-base.mk
@@ -1,5 +1,6 @@
 TARGET_SYSTEM_PROP := device/phh/treble/system.prop $(TARGET_SYSTEM_PROP)
 BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR := device/phh/treble/bluetooth
+TARGET_USERIMAGES_SPARSE_EXT_DISABLED := true
 TARGET_EXFAT_DRIVER := exfat
 DEVICE_FRAMEWORK_MANIFEST_FILE := device/phh/treble/framework_manifest.xml
 
-- 
2.25.1

