From 1c446de4c7f390e699543a440b1b71588a249c43 Mon Sep 17 00:00:00 2001
From: sooti <sooti85@gmail.com>
Date: Tue, 20 Jul 2021 13:51:24 +0300
Subject: [PATCH 36/80] Fix hotword for redmi 9S/Pro/Max

---
 rw-system.sh | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/rw-system.sh b/rw-system.sh
index 3e05358..5d1550a 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -770,6 +770,17 @@ if [ -f /system/phh/secure ];then
     setprop ctl.restart adbd
 fi
 
+# Fix the google hotword detection for android 11 on redmi note 9s
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/curtana; then
+     resetprop_phh ro.product.model "Note 9S"
+fi
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/joyeuse; then
+     resetprop_phh ro.product.model "Note 9S Pro"
+fi
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/excalibur; then
+     resetprop_phh ro.product.model "Note 9S Pro Max"
+fi
+
 for abi in "" 64;do
     f=/vendor/lib$abi/libstagefright_foundation.so
     if [ -f "$f" ];then
-- 
2.25.1

