From c18b642ba4aab23935ac0c6790866aad033000a8 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Fri, 8 Oct 2021 20:26:33 +0000
Subject: [PATCH 40/80] system.prop: set ro.odm.camera=1 for all

  !!! EXPERIMENTAL !!!
---
 system.prop | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/system.prop b/system.prop
index 3dbe6df..8b4032e 100644
--- a/system.prop
+++ b/system.prop
@@ -4,6 +4,9 @@ ro.adb.secure=0
 ro.sys.sdcardfs=0
 persist.sys.overlay.nightmode=true
 
+#experimental
+ro.odm.camera=1
+
 #Disable debugging strict mode toasts
 persist.sys.strictmode.disable=true
 persist.sys.phh.mainkeys=0
-- 
2.25.1

