From 7a8770e51e2ec2b999bde9279a6335e501be4098 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 13 Oct 2021 11:03:08 -0400
Subject: [PATCH 43/80] Don't override keymaster key if we didnt extract a spl

---
 rw-system.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index e05cdbd..14df8db 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -82,8 +82,12 @@ fixSPL() {
     if [ -n "$img" ]; then
         #Rewrite SPL/Android version if needed
         Arelease="$(getSPL "$img" android)"
+        spl="$(getSPL "$img" spl)"
         setprop ro.keymaster.xxx.release "$Arelease"
-        setprop ro.keymaster.xxx.security_patch "$(getSPL "$img" spl)"
+        setprop ro.keymaster.xxx.security_patch "$spl"
+	if [ -z "$Arelease" ] || [ -z "$spl" ];then
+		return 0
+	fi
         setprop ro.keymaster.brn Android
 
         if getprop ro.vendor.build.fingerprint |grep -qiE 'samsung.*star.*lte';then
-- 
2.25.1

