From d73e0d3de5175ee3a30aab76f30b5ad2334065c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jes=C3=BAs=20R?= <jesus.roldanr@alumno.buap.mx>
Date: Thu, 9 Dec 2021 13:32:53 -0600
Subject: [PATCH 51/80] Disable FP gestures for Motorola One Fusion Plus

---
 rw-system.sh | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/rw-system.sh b/rw-system.sh
index da8dec2..002f149 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -286,6 +286,14 @@ changeKeylayout() {
         changed=true
     fi
 
+    if getprop ro.vendor.build.fingerprint | grep -iq -e motorola/liber; then
+      if [ ! -f /mnt/phh/keylayout/uinput_nav.kl ]; then
+        cp /system/phh/empty /mnt/phh/keylayout/uinput_nav.kl
+        chmod 0644 /mnt/phh/keylayout/uinput_nav.kl
+        changed=true
+      fi
+    fi
+
     if [ "$changed" = true ]; then
         mount -o bind /mnt/phh/keylayout /system/usr/keylayout
         restorecon -R /system/usr/keylayout
-- 
2.25.1

