From 21e5c6f74ffcf0bb56e05bcdb40ad36f0bfed8d0 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sun, 19 Dec 2021 08:04:39 +0000
Subject: [PATCH 53/80] rw-system: re-factoring changeKeylayout

---
 rw-system.sh | 124 +++++++++++++++++++++++++--------------------------
 1 file changed, 61 insertions(+), 63 deletions(-)

diff --git a/rw-system.sh b/rw-system.sh
index 5110ae3..d88b4ec 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -137,32 +137,31 @@ fixSPL() {
 
 changeKeylayout() {
     mpk="/mnt/phh/keylayout"
-    cp -a /system/usr/keylayout /mnt/phh/keylayout
+    cp -a /system/usr/keylayout ${mpk}
     changed=false
     if grep -q vendor.huawei.hardware.biometrics.fingerprint /vendor/etc/vintf/manifest.xml; then
+        cp /system/phh/huawei/fingerprint.kl ${mpk}/fingerprint.kl
+        chmod 0644 ${mpk}/fingerprint.kl
         changed=true
-        cp /system/phh/huawei/fingerprint.kl /mnt/phh/keylayout/fingerprint.kl
-        chmod 0644 /mnt/phh/keylayout/fingerprint.kl
     fi
 
     if getprop ro.vendor.build.fingerprint |
         grep -qE -e "^samsung"; then
+        cp /system/phh/samsung-gpio_keys.kl ${mpk}/gpio_keys.kl
+        cp /system/phh/samsung-sec_touchscreen.kl ${mpk}/sec_touchscreen.kl
+        cp /system/phh/samsung-sec_touchkey.kl ${mpk}/sec_touchkey.kl
+        chmod 0644 ${mpk}/gpio_keys.kl ${mpk}/sec_touchscreen.kl
         changed=true
-
-        cp /system/phh/samsung-gpio_keys.kl /mnt/phh/keylayout/gpio_keys.kl
-        cp /system/phh/samsung-sec_touchscreen.kl /mnt/phh/keylayout/sec_touchscreen.kl
-        cp /system/phh/samsung-sec_touchkey.kl /mnt/phh/keylayout/sec_touchkey.kl
-        chmod 0644 /mnt/phh/keylayout/gpio_keys.kl /mnt/phh/keylayout/sec_touchscreen.kl
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -iq -e xiaomi/daisy; then
         cp /system/phh/daisy-buttonJack.kl ${mpk}/msm8953-snd-card-mtp_Button_Jack.kl
         changed=true
-        if [ ! -f /mnt/phh/keylayout/uinput-goodix.kl ]; then
+        if [ ! -f ${mpk}/uinput-goodix.kl ]; then
            cp /system/phh/daisy-uinput-goodix.kl ${mpk}/uinput-goodix.kl
            changed=true
         fi
-        if [ ! -f /mnt/phh/keylayout/uinput-fpc.kl ]; then
+        if [ ! -f ${mpk}/uinput-fpc.kl ]; then
            cp /system/phh/daisy-uinput-fpc.kl ${mpk}/uinput-fpc.kl
            changed=true
         fi
@@ -171,56 +170,55 @@ changeKeylayout() {
 
     if getprop ro.vendor.build.fingerprint | grep -iq \
         -e poco/ -e redmi/ -e xiaomi/ ; then
-        if [ ! -f /mnt/phh/keylayout/uinput-goodix.kl ]; then
-          cp /system/phh/empty /mnt/phh/keylayout/uinput-goodix.kl
-          chmod 0644 /mnt/phh/keylayout/uinput-goodix.kl
+        if [ ! -f ${mpk}/uinput-goodix.kl ]; then
+          cp /system/phh/empty ${mpk}/uinput-goodix.kl
+          chmod 0644 ${mpk}/uinput-goodix.kl
           changed=true
         fi
-        if [ ! -f /mnt/phh/keylayout/uinput-fpc.kl ]; then
-          cp /system/phh/empty /mnt/phh/keylayout/uinput-fpc.kl
-          chmod 0644 /mnt/phh/keylayout/uinput-fpc.kl
+        if [ ! -f ${mpk}/uinput-fpc.kl ]; then
+          cp /system/phh/empty ${mpk}/uinput-fpc.kl
+          chmod 0644 ${mpk}/uinput-fpc.kl
           changed=true
         fi
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -iq -e xiaomi/renoir; then
-        mpk="/mnt/phh/keylayout"
         cp /system/phh/daisy-buttonJack.kl ${mpk}/lahaina-shimaidp-snd-card_Button_Jack.kl
         chmod 0644 ${mpk}/lahaina-shimaidp-snd-card_Button_Jack.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -qi oneplus/oneplus6/oneplus6; then
-        cp /system/phh/oneplus6-synaptics_s3320.kl /mnt/phh/keylayout/synaptics_s3320.kl
-        chmod 0644 /mnt/phh/keylayout/synaptics_s3320.kl
+        cp /system/phh/oneplus6-synaptics_s3320.kl ${mpk}/synaptics_s3320.kl
+        chmod 0644 ${mpk}/synaptics_s3320.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -iq -e iaomi/perseus -e iaomi/cepheus; then
-        cp /system/phh/mimix3-gpio-keys.kl /mnt/phh/keylayout/gpio-keys.kl
-        chmod 0644 /mnt/phh/keylayout/gpio-keys.kl
+        cp /system/phh/mimix3-gpio-keys.kl ${mpk}/gpio-keys.kl
+        chmod 0644 ${mpk}/gpio-keys.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -iq -E -e '^Sony/G834'; then
-        cp /system/phh/sony-gpio-keys.kl /mnt/phh/keylayout/gpio-keys.kl
-        chmod 0644 /mnt/phh/keylayout/gpio-keys.kl
+        cp /system/phh/sony-gpio-keys.kl ${mpk}/gpio-keys.kl
+        chmod 0644 ${mpk}/gpio-keys.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint |grep -iq -E -e '^Nokia/Panther';then
-        cp /system/phh/nokia-soc_gpio_keys.kl /mnt/phh/keylayout/soc_gpio_keys.kl
-        chmod 0644 /mnt/phh/keylayout/soc_gpio_keys.kl
+        cp /system/phh/nokia-soc_gpio_keys.kl ${mpk}/soc_gpio_keys.kl
+        chmod 0644 ${mpk}/soc_gpio_keys.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint |grep -iq -E -e '^Lenovo/' && [ -f /sys/devices/virtual/touch/tp_dev/gesture_on ];then
-        cp /system/phh/lenovo-synaptics_dsx.kl /mnt/phh/keylayout/synaptics_dsx.kl
-        chmod 0644 /mnt/phh/keylayout/synaptics_dsx.kl
+        cp /system/phh/lenovo-synaptics_dsx.kl ${mpk}/synaptics_dsx.kl
+        chmod 0644 ${mpk}/synaptics_dsx.kl
         changed=true
         if ! getprop ro.vendor.build.fingerprint |grep -iq -E -e '^Lenovo/jd20'; then
-          cp /system/phh/lenovo-synaptics_dsx.kl /mnt/phh/keylayout/fts_ts.kl
-          chmod 0644 /mnt/phh/keylayout/fts_ts.kl
+          cp /system/phh/lenovo-synaptics_dsx.kl ${mpk}/fts_ts.kl
+          chmod 0644 ${mpk}/fts_ts.kl
           changed=true
         fi
     fi
@@ -228,81 +226,81 @@ changeKeylayout() {
     if ( getprop ro.build.overlay.deviceid |grep -q -e RMX1931 -e RMX1941 -e CPH1859 -e CPH1861 -e RMX2185) ||
 	    ( grep -q OnePlus /odm/etc/$(getprop ro.boot.prjname)/*.prop);then
 	echo 1 > /proc/touchpanel/double_tap_enable
-        cp /system/phh/oppo-touchpanel.kl /mnt/phh/keylayout/touchpanel.kl
-	cp /system/phh/oppo-touchpanel.kl /mnt/phh/keylayout/mtk-tpd.kl
-        chmod 0644 /mnt/phh/keylayout/touchpanel.kl
-	chmod 0644 /mnt/phh/keylayout/mtk-tpd.kl
+        cp /system/phh/oppo-touchpanel.kl ${mpk}/touchpanel.kl
+	cp /system/phh/oppo-touchpanel.kl ${mpk}/mtk-tpd.kl
+        chmod 0644 ${mpk}/touchpanel.kl
+	chmod 0644 ${mpk}/mtk-tpd.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint |grep -q -e google/;then
-        cp /system/phh/google-uinput-fpc.kl /mnt/phh/keylayout/uinput-fpc.kl
-        chmod 0644 /mnt/phh/keylayout/uinput-fpc.kl
+        cp /system/phh/google-uinput-fpc.kl ${mpk}/uinput-fpc.kl
+        chmod 0644 ${mpk}/uinput-fpc.kl
         changed=true
     fi
 
     if getprop ro.product.vendor.manufacturer |grep -q -e motorola;then
-        cp /system/phh/moto-uinput-egis.kl /mnt/phh/keylayout/uinput-egis.kl
-        cp /system/phh/moto-uinput-egis.kl /mnt/phh/keylayout/uinput-fpc.kl
-        chmod 0644 /mnt/phh/keylayout/uinput-egis.kl
-        chmod 0644 /mnt/phh/keylayout/uinput-fpc.kl
+        cp /system/phh/moto-uinput-egis.kl ${mpk}/uinput-egis.kl
+        cp /system/phh/moto-uinput-egis.kl ${mpk}/uinput-fpc.kl
+        chmod 0644 ${mpk}/uinput-egis.kl
+        chmod 0644 ${mpk}/uinput-fpc.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint |grep -q -e nubia/NX659;then
-        cp /system/phh/nubia-nubia_synaptics_dsx.kl /mnt/phh/keylayout/nubia_synaptics_dsx.kl
-        chmod 0644 /mnt/phh/keylayout/nubia_synaptics_dsx.kl
+        cp /system/phh/nubia-nubia_synaptics_dsx.kl ${mpk}/nubia_synaptics_dsx.kl
+        chmod 0644 ${mpk}/nubia_synaptics_dsx.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint |grep -q -i -e Teracube/Teracube_2e;then
-        cp /system/phh/teracube2e-mtk-kpd.kl /mnt/phh/keylayout/mtk-kpd.kl
-        chmod 0644 /mnt/phh/keylayout/mtk-kpd.kl
+        cp /system/phh/teracube2e-mtk-kpd.kl ${mpk}/mtk-kpd.kl
+        chmod 0644 ${mpk}/mtk-kpd.kl
         changed=true
     fi
 
     if getprop ro.vendor.asus.build.fp |grep -q ASUS_I01WD;then
-        cp /system/phh/zf6-goodixfp.kl /mnt/phh/keylayout/goodixfp.kl
-        cp /system/phh/zf6-googlekey_input.kl /mnt/phh/keylayout/googlekey_input.kl
-        chmod 0644 /mnt/phh/keylayout/goodixfp.kl
-        chmod 0644 /mnt/phh/keylayout/googlekey_input.kl
+        cp /system/phh/zf6-goodixfp.kl ${mpk}/goodixfp.kl
+        cp /system/phh/zf6-googlekey_input.kl ${mpk}/googlekey_input.kl
+        chmod 0644 ${mpk}/goodixfp.kl
+        chmod 0644 ${mpk}/googlekey_input.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -q -e Unihertz/;then
-        cp /system/phh/unihertz-mtk-kpd.kl /mnt/phh/keylayout/mtk-kpd.kl
-        cp /system/phh/unihertz-mtk-tpd.kl /mnt/phh/keylayout/mtk-tpd.kl
-        cp /system/phh/unihertz-mtk-tpd-kpd.kl /mnt/phh/keylayout/mtk-tpd-kpd.kl
-        cp /system/phh/unihertz-fingerprint_key.kl /mnt/phh/keylayout/fingerprint_key.kl
-        chmod 0644 /mnt/phh/keylayout/mtk-kpd.kl
-        chmod 0644 /mnt/phh/keylayout/mtk-tpd.kl
-        chmod 0644 /mnt/phh/keylayout/mtk-tpd-kpd.kl
-        chmod 0644 /mnt/phh/keylayout/fingerprint_key.kl
+        cp /system/phh/unihertz-mtk-kpd.kl ${mpk}/mtk-kpd.kl
+        cp /system/phh/unihertz-mtk-tpd.kl ${mpk}/mtk-tpd.kl
+        cp /system/phh/unihertz-mtk-tpd-kpd.kl ${mpk}/mtk-tpd-kpd.kl
+        cp /system/phh/unihertz-fingerprint_key.kl ${mpk}/fingerprint_key.kl
+        chmod 0644 ${mpk}/mtk-kpd.kl
+        chmod 0644 ${mpk}/mtk-tpd.kl
+        chmod 0644 ${mpk}/mtk-tpd-kpd.kl
+        chmod 0644 ${mpk}/fingerprint_key.kl
         changed=true
     fi
 
     if getprop ro.product.vendor.device |grep -qi mfh505glm; then
-        cp /system/phh/empty /mnt/phh/keylayout/uinput-fpc.kl
-        chmod 0644 /mnt/phh/keylayout/uinput-fpc.kl
+        cp /system/phh/empty ${mpk}/uinput-fpc.kl
+        chmod 0644 ${mpk}/uinput-fpc.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -q -i -e Blackview/BV9500Plus;then
-        cp /system/phh/bv9500plus-mtk-kpd.kl /mnt/phh/keylayout/mtk-kpd.kl
-        chmod 0644 /mnt/phh/keylayout/mtk-kpd.kl
+        cp /system/phh/bv9500plus-mtk-kpd.kl ${mpk}/mtk-kpd.kl
+        chmod 0644 ${mpk}/mtk-kpd.kl
         changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -iq -e motorola/liber; then
-      if [ ! -f /mnt/phh/keylayout/uinput_nav.kl ]; then
-        cp /system/phh/empty /mnt/phh/keylayout/uinput_nav.kl
-        chmod 0644 /mnt/phh/keylayout/uinput_nav.kl
+      if [ ! -f ${mpk}/uinput_nav.kl ]; then
+        cp /system/phh/empty ${mpk}/uinput_nav.kl
+        chmod 0644 ${mpk}/uinput_nav.kl
         changed=true
       fi
     fi
 
     if [ "$changed" = true ]; then
-        mount -o bind /mnt/phh/keylayout /system/usr/keylayout
+        mount -o bind ${mpk} /system/usr/keylayout
         restorecon -R /system/usr/keylayout
     fi
 }
-- 
2.25.1

