From 1bc0fe7e4ec59856b3c1df74d5b80d7201a93dc9 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Thu, 20 Jan 2022 08:13:47 +0000
Subject: [PATCH 58/80] try to add support for RMX2195

---
 rw-system.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index cb3578c..374678e 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -453,7 +453,8 @@ if getprop ro.build.overlay.deviceid |grep -q -e CPH1859 -e CPH1861 -e RMX1811 -
     setprop persist.sys.qcom-brightness "$(cat /sys/class/leds/lcd-backlight/max_brightness)"
 fi
 
-if getprop ro.build.overlay.deviceid |grep -iq -e RMX2020 -e RMX2027 -e RMX2040 -e RMX2193 -e RMX2191;then	
+if getprop ro.build.overlay.deviceid |grep -iq -e RMX2020 -e RMX2027 -e RMX2040 \
+    -e RMX2193 -e RMX2191 -e RMX2195; then	
     setprop persist.sys.qcom-brightness 2047
     setprop persist.sys.overlay.devinputjack true
     setprop persist.sys.phh.fingerprint.nocleanup true
-- 
2.25.1

