From 908d73cb6723f457cf5db9797e6814e52a6aaec2 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Thu, 6 Apr 2023 13:51:08 +0200
Subject: [PATCH 2/2] Support new phone model

---
 rw-system.sh | 356 +++++++++++++++++++++++++++++++++++++--------------
 1 file changed, 262 insertions(+), 94 deletions(-)

diff --git a/rw-system.sh b/rw-system.sh
index f9569dc..391cd64 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -70,6 +70,12 @@ if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/curtana -e redmi/joyeu
     setprop debug.sf.latch_unsignaled 1
     setprop debug.sf.enable_gl_backpressure 1
 fi
+    
+if getprop ro.vendor.build.fingerprint | grep -q -i -e Blackview/BV9500Plus;then
+    setprop persist.adb.nonblocking_ffs true
+else
+    setprop persist.adb.nonblocking_ffs false
+fi
 
 fixSPL() {
     if [ "$(getprop ro.product.cpu.abi)" = "armeabi-v7a" ]; then
@@ -88,6 +94,14 @@ fixSPL() {
 	if [ -z "$Arelease" ] || [ -z "$spl" ];then
 		return 0
 	fi
+
+    # Found on Cubot Pocket 3: trustkernel work only on stock model name or AOSP GSI model name
+    if [ -f /vendor/bin/hw/android.hardware.keymaster@4.1-service.trustkernel ] && [ -f /proc/tkcore/tkcore_log ];then
+        setprop debug.phh.props.teed keymaster
+        # Process name is android.hardware.keymaster@4.1-service.trustkernel
+        setprop debug.phh.props.ice.trustkernel keymaster
+    fi
+
         setprop ro.keymaster.brn Android
 
         if getprop ro.vendor.build.fingerprint |grep -qiE 'samsung.*star.*lte';then
@@ -131,6 +145,7 @@ fixSPL() {
 }
 
 changeKeylayout() {
+    mpk="/mnt/phh/keylayout"
     cp -a /system/usr/keylayout /mnt/phh/keylayout
     changed=false
     if grep -q vendor.huawei.hardware.biometrics.fingerprint /vendor/etc/vintf/manifest.xml; then
@@ -149,8 +164,21 @@ changeKeylayout() {
         chmod 0644 /mnt/phh/keylayout/gpio_keys.kl /mnt/phh/keylayout/sec_touchscreen.kl
     fi
 
+    if getprop ro.vendor.build.fingerprint | grep -iq \
+        -e poco/ -e POCO/ -e redmi/ -e xiaomi/ ; then
+        if [ ! -f /mnt/phh/keylayout/uinput-goodix.kl ]; then
+          cp /system/phh/empty /mnt/phh/keylayout/uinput-goodix.kl
+          chmod 0644 /mnt/phh/keylayout/uinput-goodix.kl
+          changed=true
+        fi
+        if [ ! -f /mnt/phh/keylayout/uinput-fpc.kl ]; then
+          cp /system/phh/empty /mnt/phh/keylayout/uinput-fpc.kl
+          chmod 0644 /mnt/phh/keylayout/uinput-fpc.kl
+          changed=true
+        fi
+    fi
+
     if getprop ro.vendor.build.fingerprint | grep -iq -e xiaomi/daisy; then
-        mpk="/mnt/phh/keylayout"
         cp /system/phh/daisy-buttonJack.kl ${mpk}/msm8953-snd-card-mtp_Button_Jack.kl
         changed=true
         if [ ! -f /mnt/phh/keylayout/uinput-goodix.kl ]; then
@@ -164,18 +192,11 @@ changeKeylayout() {
         chmod 0644 ${mpk}/uinput* ${mpk}/msm8953*
     fi
 
-    if getprop ro.vendor.build.fingerprint | grep -iq \
-        -e poco/ -e redmi/ -e xiaomi/ ; then
-        if [ ! -f /mnt/phh/keylayout/uinput-goodix.kl ]; then
-          cp /system/phh/empty /mnt/phh/keylayout/uinput-goodix.kl
-          chmod 0644 /mnt/phh/keylayout/uinput-goodix.kl
-          changed=true
-        fi
-        if [ ! -f /mnt/phh/keylayout/uinput-fpc.kl ]; then
-          cp /system/phh/empty /mnt/phh/keylayout/uinput-fpc.kl
-          chmod 0644 /mnt/phh/keylayout/uinput-fpc.kl
-          changed=true
-        fi
+    if getprop ro.vendor.build.fingerprint | grep -iq -e xiaomi/renoir; then
+        mpk="/mnt/phh/keylayout"
+        cp /system/phh/daisy-buttonJack.kl ${mpk}/lahaina-shimaidp-snd-card_Button_Jack.kl
+        chmod 0644 ${mpk}/lahaina-shimaidp-snd-card_Button_Jack.kl
+        changed=true
     fi
 
     if getprop ro.vendor.build.fingerprint | grep -qi oneplus/oneplus6/oneplus6; then
@@ -190,7 +211,7 @@ changeKeylayout() {
         changed=true
     fi
 
-    if getprop ro.vendor.build.fingerprint | grep -iq -E -e '^Sony/G834'; then
+    if getprop ro.vendor.build.fingerprint | grep -iq -E -e '^Sony/'; then
         cp /system/phh/sony-gpio-keys.kl /mnt/phh/keylayout/gpio-keys.kl
         chmod 0644 /mnt/phh/keylayout/gpio-keys.kl
         changed=true
@@ -205,12 +226,9 @@ changeKeylayout() {
     if getprop ro.vendor.build.fingerprint |grep -iq -E -e '^Lenovo/' && [ -f /sys/devices/virtual/touch/tp_dev/gesture_on ];then
         cp /system/phh/lenovo-synaptics_dsx.kl /mnt/phh/keylayout/synaptics_dsx.kl
         chmod 0644 /mnt/phh/keylayout/synaptics_dsx.kl
+        cp /system/phh/lenovo-synaptics_dsx.kl /mnt/phh/keylayout/fts_ts.kl
+        chmod 0644 /mnt/phh/keylayout/fts_ts.kl
         changed=true
-        if ! getprop ro.vendor.build.fingerprint |grep -iq -E -e '^Lenovo/jd20'; then
-          cp /system/phh/lenovo-synaptics_dsx.kl /mnt/phh/keylayout/fts_ts.kl
-          chmod 0644 /mnt/phh/keylayout/fts_ts.kl
-          changed=true
-        fi
     fi
 
     if ( getprop ro.build.overlay.deviceid |grep -q -e RMX1931 -e RMX1941 -e CPH1859 -e CPH1861 -e RMX2185) ||
@@ -269,12 +287,41 @@ changeKeylayout() {
         changed=true
     fi
 
-    if getprop ro.product.vendor.device |grep -qi mfh505glm; then
+    if getprop ro.vendor.build.fingerprint | grep -q -i -e Blackview/BV9500Plus;then
+        cp /system/phh/bv9500plus-mtk-kpd.kl /mnt/phh/keylayout/mtk-kpd.kl
+        chmod 0644 /mnt/phh/keylayout/mtk-kpd.kl
+        changed=true
+    fi
+
+    if getprop ro.product.vendor.device |grep -qi -e mfh505glm -e fh50lm; then
         cp /system/phh/empty /mnt/phh/keylayout/uinput-fpc.kl
         chmod 0644 /mnt/phh/keylayout/uinput-fpc.kl
         changed=true
     fi
 
+    if getprop ro.vendor.build.fingerprint | grep -iq -e motorola/liber; then
+        cp /system/phh/moto-liber-gpio-keys.kl /mnt/phh/keylayout/gpio-keys.kl
+        chmod 0644 /mnt/phh/keylayout/gpio-keys.kl
+
+        cp /system/phh/empty /mnt/phh/keylayout/uinput_nav.kl
+        chmod 0644 /mnt/phh/keylayout/uinput_nav.kl
+        changed=true
+    fi
+
+    if getprop ro.vendor.build.fingerprint | grep -iq DOOGEE/S88Pro;then
+          cp /system/phh/empty /mnt/phh/keylayout/sf-keys.kl
+          chmod 0644 /mnt/phh/keylayout/sf-keys.kl
+          changed=true
+    fi
+
+    if getprop ro.vendor.build.fingerprint |grep -iq tecno/kd7;then
+        # Enable dt2w
+        echo cc1 > /proc/gesture_function
+        cp /system/phh/tecno-touchpanel.kl /mnt/phh/keylayout/mtk-tpd.kl
+        chmod 0644 /mnt/phh/keylayout/mtk-tpd.kl
+        changed=true
+    fi
+
     if [ "$changed" = true ]; then
         mount -o bind /mnt/phh/keylayout /system/usr/keylayout
         restorecon -R /system/usr/keylayout
@@ -293,11 +340,11 @@ fi
 if mount -o remount,rw /system; then
     resize2fs "$(grep ' /system ' /proc/mounts | cut -d ' ' -f 1)" || true
 else
-    remount system
     mount -o remount,rw /
     major="$(stat -c '%D' /.|sed -E 's/^([0-9a-f]+)([0-9a-f]{2})$/\1/g')"
     minor="$(stat -c '%D' /.|sed -E 's/^([0-9a-f]+)([0-9a-f]{2})$/\2/g')"
     mknod /dev/tmp-phh b $((0x$major)) $((0x$minor))
+    blockdev --setrw /dev/tmp-phh
     resize2fs /dev/root || true
     resize2fs /dev/tmp-phh || true
 fi
@@ -328,15 +375,14 @@ if grep vendor.huawei.hardware.biometrics.fingerprint /vendor/manifest.xml; then
 fi
 
 foundFingerprint=false
-for manifest in /vendor/manifest.xml /vendor/etc/vintf/manifest.xml /odm/etc/vintf/manifest.xml;do
-    if grep -q \
-            -e android.hardware.biometrics.fingerprint \
-            -e vendor.oppo.hardware.biometrics.fingerprint \
-            -e vendor.oplus.hardware.biometrics.fingerprint \
-            $manifest;
-        then
-        foundFingerprint=true
-    fi
+for manifest in /vendor/manifest.xml /vendor/etc/vintf /odm/etc/vintf;do
+	if grep -q \
+		-e android.hardware.biometrics.fingerprint \
+		-e vendor.oppo.hardware.biometrics.fingerprint \
+		-e vendor.oplus.hardware.biometrics.fingerprint \
+		-r $manifest;then
+			foundFingerprint=true
+	fi
 done
 
 if [ "$foundFingerprint" = false ];then
@@ -364,9 +410,9 @@ if getprop ro.vendor.build.fingerprint | grep -iq \
     setprop persist.sys.qcom-brightness -1
 fi
 
-# Lenovo Z5s brightness flickers without this setting
+# Lenovo Z5s & Xiaomi Mi10TLite brightness flickers without this setting
 if getprop ro.vendor.build.fingerprint | grep -iq \
-    -e Lenovo/jd2019; then
+    -e Lenovo/jd2019 -e Xiaomi/gauguin -e Redmi/gauguin; then
     setprop persist.sys.qcom-brightness -1
 fi
 
@@ -396,7 +442,7 @@ if getprop ro.vendor.build.fingerprint | grep -q -i \
     -e xiaomi/nitrogen -e xiaomi/whyred -e xiaomi/platina \
     -e xiaomi/ysl -e nubia/nx60 -e nubia/nx61 -e xiaomi/tulip \
     -e xiaomi/lavender -e xiaomi/olive -e xiaomi/olivelite -e xiaomi/pine \
-    -e Redmi/lancelot -e Redmi/galahad -e Redmi/merlin -e Redmi/angelican -e Redmi/dandelion; then
+    -e Redmi/lancelot -e Redmi/galahad -e POCO/evergreen; then
     setprop persist.sys.qcom-brightness "$(cat /sys/class/leds/lcd-backlight/max_brightness)"
 fi
 
@@ -429,7 +475,8 @@ if getprop ro.build.overlay.deviceid |grep -q -e CPH1859 -e CPH1861 -e RMX1811 -
     setprop persist.sys.qcom-brightness "$(cat /sys/class/leds/lcd-backlight/max_brightness)"
 fi
 
-if getprop ro.build.overlay.deviceid |grep -iq -e RMX2020 -e RMX2027 -e RMX2040 -e RMX2193 -e RMX2191;then	
+if getprop ro.build.overlay.deviceid |grep -iq -e RMX2020 -e RMX2027 -e RMX2040 -e RMX2193 \
+    -e RMX2193 -e RMX2191 -e RMX2195;then
     setprop persist.sys.qcom-brightness 2047
     setprop persist.sys.overlay.devinputjack true
     setprop persist.sys.phh.fingerprint.nocleanup true
@@ -443,8 +490,7 @@ if getprop ro.vendor.build.fingerprint | grep -iq \
     -e motorola/hannah -e motorola/james -e motorola/pettyl -e xiaomi/cepheus \
     -e xiaomi/grus -e xiaomi/cereus -e xiaomi/cactus -e xiaomi/raphael -e xiaomi/davinci \
     -e xiaomi/ginkgo -e xiaomi/willow -e xiaomi/laurel_sprout -e xiaomi/andromeda \
-    -e iaomi/renoir \
-    -e redmi/curtana -e -e redmi/joyeuse -e redmi/excalibur -e redmi/picasso \
+    -e xiaomi/gauguin -e redmi/gauguin -e redmi/curtana -e redmi/picasso \
     -e bq/Aquaris_M10 ; then
     mount -o bind /mnt/phh/empty_dir /vendor/lib64/soundfx
     mount -o bind /mnt/phh/empty_dir /vendor/lib/soundfx
@@ -546,11 +592,6 @@ if busybox_phh unzip -p /vendor/app/ims/ims.apk classes.dex | grep -qF -e Landro
     mount -o bind /system/phh/empty /vendor/app/ims/ims.apk
 fi
 
-#experimental
-if [ ! getprop ro.board.platform |grep -q mt6785 ] && [ ! grep -q 338 /sys/devices/soc0/soc_id ]; then
-    setprop debug.sf.latch_unsignaled 1
-fi
-
 if getprop ro.hardware | grep -qF exynos; then
     setprop debug.sf.latch_unsignaled 1
 fi
@@ -565,31 +606,31 @@ fi
 
 if getprop ro.vendor.build.fingerprint | grep -iq -e xiaomi/daisy; then
     setprop debug.sf.latch_unsignaled 1
-    setprop ro.telephony.default_network 9,9
-    setprop persist.vendor.radio.block_allow_data 0
-    setprop telephony.lteOnGsmDevice 1
+    setprop debug.sf.enable_hwc_vds 1
 fi
 
-if getprop ro.vendor.build.fingerprint | grep -iq -e xiaomi/lavender; then
+if getprop ro.vendor.build.fingerprint | grep -iq -e Redmi/merlin; then
     setprop debug.sf.latch_unsignaled 1
-    setprop ro.netflix.bsp_rev Q660-13149-1
-    setprop ro.telephony.default_network 9,9
-    setprop persist.vendor.radio.block_allow_data 0
-    setprop telephony.lteOnGsmDevice 1
+    setprop debug.sf.enable_hwc_vds 0
 fi
 
-if getprop ro.vendor.build.fingerprint | grep -iq -e Redmi/merlin; then
+if getprop ro.vendor.build.fingerprint | grep -iq -e Redmi/rosemary \
+    -e Redmi/secret -e Redmi/maltose; then
     setprop debug.sf.latch_unsignaled 1
     setprop debug.sf.enable_hwc_vds 0
+
+    # Exclude FP input devices
+    mount -o bind /system/phh/rosemary-excluded-input-devices.xml /system/etc/excluded-input-devices.xml
 fi
 
 # Iceows patch - NFC NXP conf remove, you must set by specify model patch
 if getprop ro.vendor.build.fingerprint | grep -iq -E -e 'huawei|honor' || getprop persist.sys.overlay.huawei | grep -iq -E -e 'true'; then
-    setprop debug.sf.latch_unsignaled 1
+
+   setprop debug.sf.latch_unsignaled 1
 
     # Set proper /sdcard permissions to avoid dead storage
-    chown media_rw:media_rw /data/media/0
-    chmod 0770 /data/media/0
+   chown media_rw:media_rw /data/media/0
+   chmod 0770 /data/media/0
 
     # Disable this watchlss extension to fix logspams and dead gps
     mount -o bind /system/phh/empty /vendor/bin/gnss_watchlssd_thirdparty
@@ -613,6 +654,7 @@ if getprop ro.vendor.build.fingerprint | grep -qE -e ".*(crown|star)[q2]*lte.*"
     done
 fi
 
+
 # This matches both Razer Phone 1 & 2
 if getprop ro.vendor.build.fingerprint |grep -qE razer/cheryl;then
 	setprop ro.audio.monitorRotation true
@@ -632,10 +674,6 @@ if getprop ro.vendor.build.fingerprint | grep -qiE '^samsung'; then
     fi
 fi
 
-if getprop ro.vendor.build.fingerprint | grep -qiE '^samsung/j4lteub'; then
-    setprop vendor.camera.hal1.packagelist com.whatsapp,org.telegram.messenger
-fi
-
 if getprop ro.vendor.build.fingerprint | grep -qE '^xiaomi/wayne/wayne.*'; then
     # Fix camera on DND, ugly workaround but meh
     setprop audio.camerasound.force true
@@ -726,7 +764,6 @@ fi
 
 setprop ctl.stop console
 dmesg -n 1
-
 copyprop() {
     p="$(getprop "$2")"
     if [ "$p" ]; then
@@ -734,34 +771,33 @@ copyprop() {
     fi
 }
 
-copyprop ro.build.device ro.vendor.build.device
-copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
-copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
-copyprop ro.build.fingerprint ro.vendor.build.fingerprint
-copyprop ro.build.device ro.vendor.product.device
-copyprop ro.product.system.device ro.vendor.product.device
-copyprop ro.product.device ro.vendor.product.device
-copyprop ro.product.system.device ro.product.vendor.device
-copyprop ro.product.device ro.product.vendor.device
-copyprop ro.product.system.name ro.vendor.product.name
-copyprop ro.product.name ro.vendor.product.name
-copyprop ro.product.system.name ro.product.vendor.name
-copyprop ro.product.name ro.product.vendor.name
-copyprop ro.system.product.brand ro.vendor.product.brand
-copyprop ro.product.brand ro.vendor.product.brand
-copyprop ro.product.system.model ro.vendor.product.model
-copyprop ro.product.model ro.vendor.product.model
-copyprop ro.product.system.model ro.product.vendor.model
-copyprop ro.product.model ro.product.vendor.model
-copyprop ro.build.product ro.vendor.product.model
-copyprop ro.build.product ro.product.vendor.model
-copyprop ro.system.product.manufacturer ro.vendor.product.manufacturer
-copyprop ro.product.manufacturer ro.vendor.product.manufacturer
-copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
-copyprop ro.product.manufacturer ro.product.vendor.manufacturer
-copyprop build.version.security_patch ro.build.version.security_patch
-
-if [ -f /system/phh/secure ];then
+if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ] || [ -f /data/adb/phh/secure ];then
+    copyprop ro.build.device ro.vendor.build.device
+    copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
+    copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
+    copyprop ro.build.fingerprint ro.vendor.build.fingerprint
+    copyprop ro.build.device ro.vendor.product.device
+    copyprop ro.product.system.device ro.vendor.product.device
+    copyprop ro.product.device ro.vendor.product.device
+    copyprop ro.product.system.device ro.product.vendor.device
+    copyprop ro.product.device ro.product.vendor.device
+    copyprop ro.product.system.name ro.vendor.product.name
+    copyprop ro.product.name ro.vendor.product.name
+    copyprop ro.product.system.name ro.product.vendor.device
+    copyprop ro.product.name ro.product.vendor.device
+    copyprop ro.system.product.brand ro.vendor.product.brand
+    copyprop ro.product.brand ro.vendor.product.brand
+    copyprop ro.product.system.model ro.vendor.product.model
+    copyprop ro.product.model ro.vendor.product.model
+    copyprop ro.product.system.model ro.product.vendor.model
+    copyprop ro.product.model ro.product.vendor.model
+    copyprop ro.build.product ro.vendor.product.model
+    copyprop ro.build.product ro.product.vendor.model
+    copyprop ro.system.product.manufacturer ro.vendor.product.manufacturer
+    copyprop ro.product.manufacturer ro.vendor.product.manufacturer
+    copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
+    copyprop ro.product.manufacturer ro.product.vendor.manufacturer
+
     (getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
         [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
     done
@@ -780,6 +816,21 @@ if [ -f /system/phh/secure ];then
 
     resetprop_phh ro.adb.secure 1
     setprop ctl.restart adbd
+
+    # Hide system/xbin/su
+    mount /mnt/phh/empty_dir /system/xbin
+    mount /mnt/phh/empty_dir /system/app/me.phh.superuser
+    mount /system/phh/empty /system/xbin/phh-su
+else
+    mkdir /mnt/phh/xbin
+    chmod 0755 /mnt/phh/xbin
+    chcon u:object_r:system_file:s0 /mnt/phh/xbin
+
+    #phh-su will bind over this empty file to make a real su
+    touch /mnt/phh/xbin/su
+    chcon u:object_r:system_file:s0 /mnt/phh/xbin/su
+
+    mount -o bind /mnt/phh/xbin /system/xbin
 fi
 
 # Fix the google hotword detection for android 11 on redmi note 9s
@@ -948,7 +999,7 @@ fi
 
 if getprop ro.vendor.build.fingerprint |grep -qiE '^samsung/';then
     for f in /sys/class/lcd/panel/actual_mask_brightness /sys/class/lcd/panel/mask_brightness /sys/class/lcd/panel/device/backlight/panel/brightness /sys/class/backlight/panel0-backlight/brightness;do
-        if [ "$(stat -c '%U' "$f")" == "root" ];then
+        if [ "$(stat -c '%U' "$f")" == "root" ] || [ "$(ls -lZ "$f" | grep -oE 'u:object_r:[^:]*:s0')" == "u:object_r:sysfs:s0" ];then
             chcon u:object_r:sysfs_lcd_writable:s0 $f
             chmod 0644 $f
             chown system:system $f
@@ -958,14 +1009,21 @@ if getprop ro.vendor.build.fingerprint |grep -qiE '^samsung/';then
     setprop persist.sys.phh.fod.samsung true
 fi
 
-if getprop ro.vendor.build.fingerprint |grep -qiE '^oneplus/';then
-    setprop persist.sys.phh.fod.bbk true
+if getprop ro.vendor.build.fingerprint | grep -q -e samsung/o1s -e samsung/t2s -e samsung/p3s; then
+    setprop persist.sys.phh.ultrasonic_udfps true
+fi
+
+if getprop ro.vendor.build.fingerprint |grep -qiE -e ASUS_I006D -e ASUS_I003;then
+	setprop persist.sys.phh.fod.asus true
 fi
-if getprop ro.build.overlay.deviceid |grep -qiE -e '^RMX' -e '^CPH';then
+
+if (getprop ro.vendor.build.fingerprint;getprop ro.odm.build.fingerprint) |grep -qiE '^oneplus/' ||
+	getprop ro.build.overlay.deviceid |grep -qiE -e '^RMX' -e '^CPH' ||
+	[ -n "$(getprop ro.separate.soft)" ];then
     setprop persist.sys.phh.fod.bbk true
 fi
 
-if getprop ro.build.overlay.deviceid |grep -iq -e RMX1941 -e RMX1945 -e RMX1943 -e RMX1942;then	
+if getprop ro.build.overlay.deviceid |grep -iq -e RMX1941 -e RMX1945 -e RMX1943 -e RMX1942;then
     setprop persist.sys.qcom-brightness "$(cat /sys/class/leds/lcd-backlight/max_brightness)"
     setprop persist.sys.phh.mainkeys 0
 fi
@@ -976,16 +1034,51 @@ fi
 
 resetprop_phh ro.bluetooth.library_name libbluetooth.so
 
+board="$(getprop ro.board.platform)"
+
 if getprop ro.vendor.build.fingerprint |grep -iq xiaomi/cepheus;then
-    setprop ro.netflix.bsp_rev Q855-16947-1
+	setprop ro.netflix.bsp_rev Q855-16947-1
+fi
+    
+if [ "$board" = atoll ] || [ "$board" = sm6250 ]; then
+	setprop ro.netflix.bsp_rev Q6250-19132-1
 fi
 
 if getprop ro.vendor.build.fingerprint |grep -qi redmi/curtana -e redmi/joyeuse -e redmi/excalibur; then
-    setprop ro.netflix.bsp_rev Q6250-19132-1
+	setprop ro.netflix.bsp_rev Q6250-19132-1
+fi
+
+if [ "$board" = msmnile ]; then
+	setprop ro.netflix.bsp_rev Q855-16947-1
 fi
 
 if getprop ro.vendor.build.fingerprint |grep -iq xiaomi/renoir;then
-    setprop ro.netflix.bsp_rev Q875-32774-1
+	setprop ro.netflix.bsp_rev Q875-32774-1
+fi
+
+if [ "$board" = sm6150 ]; then
+	setprop ro.netflix.bsp_rev Q6150-17263-1
+fi
+
+if [ "$board" = mt6768 ]; then
+	setprop ro.netflix.bsp_rev MTK6768-19055-1
+fi
+
+if [ "$board" = lahaina ]; then
+	setprop ro.netflix.bsp_rev Q875-32774-1
+	resetprop_phh ro.config.media_vol_steps 25
+	resetprop_phh ro.config.media_vol_default 15
+fi
+
+if [ "$board" = universal8825 ];then
+	setprop ro.netflix.bsp_rev EXYNOS1280-34993-1
+fi
+
+if getprop ro.vendor.build.fingerprint |grep -qi Nokia/Phoenix;then
+    setprop ro.netflix.bsp_rev Q845-05000-1
+    setprop debug.sf.latch_unsignaled 1
+    setprop sys.use_fifo_ui 1
+    setprop media.settings.xml "/vendor/etc/media_profiles_vendor.xml"
 fi
 
 # Set props for Vsmart Live's fod
@@ -1027,5 +1120,80 @@ fi
 # Disable secondary watchdogs
 echo -n V > /dev/watchdog1
 
+# Fix watchdog issue on Samsung Galaxy A20s
+if getprop ro.vendor.build.fingerprint | grep -iq samsung/a20sub/a20s; then
+    echo -n V > /dev/watchdog0
+fi
+
+if getprop ro.vendor.build.fingerprint | grep -iq samsung/a11que;then
+	echo -n V > /dev/watchdog0
+fi
+	
+if [ "$vndk" -le 30 ];then
+	# On older vendor the default behavior was to disable color management
+	# Don't override vendor value, merely add a fallback
+	setprop ro.surface_flinger.use_color_management false
+fi
+
+if [ "$(stat -c '%U'  /dev/nxp_smartpa_dev)" == "root" ] &&
+	[ "$(stat -c '%G' /dev/nxp_smartpa_dev)" == "root" ];then
+    chown root:audio /dev/nxp_smartpa_dev
+    chmod 0660 /dev/nxp_smartpa_dev
+fi
+if getprop ro.odm.build.fingerprint |grep -q Huawei/Chicago/Chicago_VTR;then
+    setprop ctl.stop aptouch
+fi
+
+if getprop ro.vendor.build.fingerprint | grep -iq -e GOME/GOME_U9; then
+    cp -a /system/etc/smartpa_params /mnt/phh/smartpa_params
+    cp /system/phh/gome/fs16xx_01s_left.preset /mnt/phh/smartpa_params/fs16xx_01s_left.preset
+    cp /system/phh/gome/fs16xx_01s_mono.preset /mnt/phh/smartpa_params/fs16xx_01s_mono.preset
+    cp /system/phh/gome/fs16xx_01s_right.preset /mnt/phh/smartpa_params/fs16xx_01s_right.preset
+    chmod 0644 /mnt/phh/smartpa_params/fs16xx_01s_left.preset
+    chmod 0644 /mnt/phh/smartpa_params/fs16xx_01s_mono.preset
+    chmod 0644 /mnt/phh/smartpa_params/fs16xx_01s_right.preset
+    mount -o bind /mnt/phh/smartpa_params /system/etc/smartpa_params
+    restorecon -R /system/etc/smartpa_params
+fi
+
+if getprop ro.vendor.build.fingerprint | grep -iq -e UMIDIGI/UMIDIGI_X; then
+    cp -a /system/etc/smartpa_params /mnt/phh/smartpa_params
+    cp /system/phh/umidigi/fs16xx_01s_mono.preset /mnt/phh/smartpa_params/fs16xx_01s_mono.preset
+    chmod 0644 /mnt/phh/smartpa_params/fs16xx_01s_mono.preset
+    mount -o bind /mnt/phh/smartpa_params /system/etc/smartpa_params
+    restorecon -R /system/etc/smartpa_params
+fi
+
+if getprop ro.vendor.build.fingerprint |grep -iq motorola/kane;then
+	setprop persist.sys.bt.esco_transport_unit_size 16
+fi
+
+if [ -f /vendor/bin/ccci_rpcd ];then
+    setprop debug.phh.props.ccci_rpcd vendor
+fi
+
+if getprop ro.vendor.build.fingerprint | grep -qi -e iaomi/mona; then
+    copyprop ro.product.manufacturer ro.product.vendor.manufacturer
+fi
+
+if getprop ro.vendor.build.fingerprint | grep -iq -e motorola/liber; then
+  cp /vendor/etc/audio_policy_configuration.xml /mnt/phh/
+  sed -i '/r_submix_audio_policy_configuration/a \t<xi:include href="/vendor/etc/a2dp_audio_policy_configuration.xml"/>' /mnt/phh/audio_policy_configuration.xml
+  mount -o bind /mnt/phh/audio_policy_configuration.xml /vendor/etc/audio_policy_configuration.xml
+  chcon -h u:object_r:vendor_configs_file:s0 /vendor/etc/audio_policy_configuration.xml
+  chmod 644 /vendor/etc/audio_policy_configuration.xml
+fi
+
+mount /system/phh/empty /vendor/etc/permissions/samsung.hardware.uwb.xml
+mount /system/phh/empty /vendor/bin/install-recovery.sh
+
+if getprop ro.vendor.radio.default_network |grep -qE '[0-9]';then
+  setprop ro.telephony.default_network $(getprop ro.vendor.radio.default_network)
+fi
+
+if getprop ro.vendor.build.fingerprint |grep -iq redmi/camellia;then
+	setprop persist.sys.qcom-brightness 4095
+fi
+
 mount -o bind /mnt/phh/empty_dir /vendor/app/qti-logkit
 mount -o bind /mnt/phh/empty_dir /vendor/app/qti-logkit-lite
-- 
2.25.1

