From 78f946937c2751d0067913ea02f3ce537a3a37b5 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Mon, 21 Feb 2022 14:01:00 +0000
Subject: [PATCH 69/80] Overwrite media volume vendor values on Xiaomi Mi 11
 Lite 5G

---
 rw-system.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rw-system.sh b/rw-system.sh
index 0428b7f..8b2198f 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -1018,6 +1018,8 @@ fi
 
 if getprop ro.vendor.build.fingerprint |grep -iq xiaomi/renoir;then
     setprop ro.netflix.bsp_rev Q875-32774-1
+    resetprop_phh ro.config.media_vol_steps 25
+    resetprop_phh ro.config.media_vol_default 15
 fi
 
 # Set props for Vsmart Live's fod
-- 
2.25.1

