From c2769872d2d9b551ddd7a39cfc03300f42cc8e28 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Mon, 21 Feb 2022 18:01:13 -0500
Subject: [PATCH 71/80] Give Sony shutter/focus on all Sony devices

---
 rw-system.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index 8b2198f..f6c7447 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -200,7 +200,7 @@ changeKeylayout() {
         changed=true
     fi
 
-    if getprop ro.vendor.build.fingerprint | grep -iq -E -e '^Sony/G834'; then
+    if getprop ro.vendor.build.fingerprint | grep -iq -E -e '^Sony/'; then
         cp /system/phh/sony-gpio-keys.kl ${mpk}/gpio-keys.kl
         chmod 0644 ${mpk}/gpio-keys.kl
         changed=true
-- 
2.25.1

