From 82f6d63f8f2c6cc1680856c2c4c1dd009d7139a7 Mon Sep 17 00:00:00 2001
From: CHIDILights <62822977+CHIDILights@users.noreply.github.com>
Date: Mon, 28 Feb 2022 00:12:32 +0900
Subject: [PATCH 72/80] Disable fingerprint gestures for Stylo 7 4G variant

---
 rw-system.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index f6c7447..a01336b 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -279,7 +279,7 @@ changeKeylayout() {
         changed=true
     fi
 
-    if getprop ro.product.vendor.device |grep -qi mfh505glm; then
+    if getprop ro.product.vendor.device |grep -qi -e mfh505glm -e fh50lm; then
         cp /system/phh/empty ${mpk}/uinput-fpc.kl
         chmod 0644 ${mpk}/uinput-fpc.kl
         changed=true
-- 
2.25.1

