From 7f0f0c8b0b91b79ce5fcdd26c565c869a66c3d06 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jes=C3=BAs=20R?= <jrr@disroot.org>
Date: Fri, 4 Mar 2022 08:48:24 -0600
Subject: [PATCH 74/80] Add line after r_submix_audio_policy_configuration

---
 rw-system.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index fe1d309..62ded07 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -1101,7 +1101,7 @@ fi
 
 if getprop ro.vendor.build.fingerprint | grep -iq -e motorola/liber; then
   cp /vendor/etc/audio_policy_configuration.xml /mnt/phh/
-  sed -i '198i <xi:include href="/vendor/etc/a2dp_audio_policy_configuration.xml"/>' /mnt/phh/audio_policy_configuration.xml
+  sed -i '/r_submix_audio_policy_configuration/a \t<xi:include href="/vendor/etc/a2dp_audio_policy_configuration.xml"/>' /mnt/phh/audio_policy_configuration.xml
   mount -o bind /mnt/phh/audio_policy_configuration.xml /vendor/etc/audio_policy_configuration.xml
   chcon -h u:object_r:vendor_configs_file:s0 /vendor/etc/audio_policy_configuration.xml
   chmod 644 /vendor/etc/audio_policy_configuration.xml
-- 
2.25.1

