From 782f697d642202f62a7f530218ba678f0b77cf00 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 22 Mar 2022 20:21:09 -0400
Subject: [PATCH 80/80] Add Samsung way to grab ro.telephony.default_network

---
 rw-system.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/rw-system.sh b/rw-system.sh
index 810d255..78ba3c9 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -1113,3 +1113,7 @@ fi
 
 mount /system/phh/empty /vendor/etc/permissions/samsung.hardware.uwb.xml
 mount /system/phh/empty /vendor/bin/install-recovery.sh
+
+if getprop ro.vendor.radio.default_network |grep -qE '[0-9]';then
+  setprop ro.telephony.default_network $(getprop ro.vendor.radio.default_network)
+fi
-- 
2.25.1

